core/src/main/java/org/pentaho/di/core/row/value/ValueMetaTimestamp.java

```diff
@@ -50,6 +51,8 @@
 import org.pentaho.di.core.row.value.timestamp.SimpleTimestampFormat;
 
 public class ValueMetaTimestamp extends ValueMetaDate {
+  private final String format = System.getProperty( Const.KETTLE_TIMESTAMP_OUTPUT_FORMAT, "NANOSECONDS" );
```

Comment:
Don't use `System.getProperty`.... remember?

Reply:
So, @pentaho-lmartins wanted to use the VariableSpace to access the property. Thus, we could define format globally, as a Kettle Property, or per-transformation, as a Kettle Parameter. However, to bring the VariableSpace (the step) into the ValueMetaTimestamp, we would have ValueMetaInterface, adding a VariableSpace argument to getNumber, getInteger and getBigDecimal. I'm not sure that we want to do that... Can you give me your thoughts @bmorrise @mbatchelor 

Now, I'm using EnvUtil instead of System directly. Nevertheless, keep in mind that getSystemProperty of EnvUtil, uses System to get the property. So I'm literally doing the same thing! :P

Reply:
Just so you know, this can be done without losing backward compatibility issues. I personally have no issue as long as the existing methods get marked as deprecated, and that they delegate to the new methods passing in a default VariableSpace or null allowing the method to continue to work as it always had in the past. That way our embedders and folks that extend the code won't suddenly break. The advantage to passing in a VariableSpace is several-fold as it allows a trans/job to override how the conversion is performed on a case-by-case basis if required by the use-case.